cmd/create.go

```diff
@@ -205,42 +207,49 @@ func createStack(ctx context.Context, cmd *cobra.Command, options *stackservice.
 	if err != nil {
 		return errors.Wrap(err, "failed to update the stack meta")
 	}
-	logrus.Infof("creating stack '%s'", options.StackName)
 
-	stack, err := options.StackService.Add(ctx, options.StackName)
+	stack, err := options.StackService.Add(ctx, options.StackName, dryRun)
 	if err != nil {
 		return errors.Wrap(err, "failed to add the stack")
 	}
+
 	logrus.Debugf("setting stackMeta %v", options.StackMeta)
 	stack = stack.WithMeta(options.StackMeta)
 
-	err = stack.Apply(ctx, getWaitOptions(options.Backend, options.StackName))
+	err = stack.Apply(ctx, getWaitOptions(options), dryRun)
 	if err != nil {
 		return errors.Wrap(err, "failed to successfully create the stack")
 	}
 
-	shouldRunMigration, err := happyCmd.ShouldRunMigrations(cmd, options.HappyConfig)
-	if err != nil {
-		return err
-	}
-	if shouldRunMigration {
-		err = runMigrate(cmd, options.StackName)
+	if dryRun {
+		logrus.Infof("cleaning up stack '%s'", options.StackName)
+		err = options.StackService.Remove(ctx, options.StackName, true)
```

Comment:
why do we need to remove if this was a dry run?

Reply:
Workspace has to be created even for a dry run, so it will be cleaned up after the dry run is complete. Workspace depends on TF variables that are not available locally, and even if they are their values can mismatch those set up by the creator workspace (i.e. `env-rdev`).

Reply:
The only other way to handle it is to create a temp TFE workspace without the stack, and then clean it up afterwards; but that leads to a more complex execution path, as today TFE workspace is created by the creator workspace (i.e. `env-rdev`)